dvc/repo/experiments/save.py

```diff
@@ -21,24 +46,41 @@ def save(
 
     Returns the saved experiment's SHAs.
     """
-    queue = repo.experiments.workspace_queue
     logger.debug("Saving workspace in %s", os.getcwd())
 
-    staged, _, _ = repo.scm.status(untracked_files="no")
-    if staged:
+    _, _, untracked = repo.scm.status()
+    if untracked:
         logger.warning(
-            "Your workspace contains staged Git changes which will be "
-            "unstaged before saving this experiment."
+            "The following untracked files were present in "
+            "the workspace before saving but "
+            "will not be included in the experiment commit:\n"
+            "\t%s",
+            ", ".join(
+                untracked
+                if not include_untracked
+                else [
+                    file for file in untracked if file not in include_untracked
+                ]
+            ),
         )
-        repo.scm.reset()
 
-    entry = repo.experiments.new(queue=queue, name=name, force=force)
-    executor = queue.init_executor(repo.experiments, entry)
+    with repo.scm.detach_head(client="dvc") as orig_head:
+        stash_head = orig_head
+        rev = repo.scm.get_rev()
+        assert stash_head == rev
```

Comment:
When could this not be True?
Should we raise a proper error instead of `assert`?

Reply:
That should never happen, cleaned this up